keps/sig-instrumentation/4200-healthcheck-metrics/README.md

```diff
@@ -0,0 +1,297 @@
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP intends to allow us to emit health check data in a structured and consistent way,
+so that monitoring agents can consume healthz/livez/readyz data and create SLOs/alerts off
+of these inputs.
+
+## Motivation
+
+Healthchecking data is currently surfaced in unstructured format and is scraped by monitoring
+agents (as well as the kubelet), which must be configured to interpret the health data and
+to act upon it. This process does not lend itself readily to the creation of availability SLOs,
+since we basically require an outside agent to parse the health endpoint and convert this signal
+into an SLI. 
+
+### Goals
+
+- Create a uniform interface by which we can consume health checking information
+- Allow availability SLIs to be created without a specialized monitoring agent
+
+
+### Non-Goals
+
+- Creation of SLOs are out of scope. This KEP specifically targets the creation of signals which can be used as SLIs.
+
+## Proposal
+
+We are proposing to add a new endpoint on Kubernetes components `/health/metrics` which returns
```

Comment:
The name should be consistent either with our existing checks (/healthz) or metrics (/metrics) as another comment noted.  The advantage of the former is that the default security mapping may be correct already (anyone who can view healthz should see these metrics), the latter is more consistent and we already have a few places where we argued different endpoints of metrics may have different authorization levels (resource metrics under scheduler expose pod names, while scheduler general metrics don’t).

I’d probably prefer /metrics/health over /healthz/metrics

Reply:
sure, sounds reasonable to me.